syntax = "proto3";

package fastmitigation;

service Fastnetmon {
    rpc GetBanlist(BanListRequest) returns (stream BanListReply) {}
    rpc ExecuteBan(ExecuteBanRequest) returns (ExecuteBanReply) {}
    rpc ExecuteUnBan(ExecuteBanRequest) returns (ExecuteBanReply) {}
}

// We could not create RPC method without params
message BanListRequest {

}

message BanListReply {
    string ip_address = 1;
}

message ExecuteBanRequest {
    string ip_address = 1;
}

message ExecuteBanReply {
    bool result = 1;
}
